pkgs/top-level/all-packages.nix

```diff
@@ -26702,14 +26704,17 @@ with pkgs;
     withHomed = false;
     withHwdb = false;
     withImportd = false;
+    withKmod = false;
```

Comment:
As far as I understand we only care about the `libudev` provided by the `systemdMinimal`, so I have disabled the `kmod` integration here, because AFAIK it is only used by the `systemd-modules-load`, but not in the Udev implementation ([this](https://github.com/systemd/systemd/blob/main/src/udev/udev-builtin-kmod.c) is used instead).

Reply:
mobile-nixos at least uses the udevd from systemdMinimal, so this change breaks mobile-nixos.

I think it's fair enough to say that systemdStage1 or a full systemd should be used instead, but in that case we should probably also prevent udevd and rules from being installed at all in the minimal (libudev-only) variant of the package -- because I imagine cases where a kmod-free `udevd` is indeed desired should be extremely rare and if someone needs that they'll probably know it.

Reply:
(the kmod-builtin thing is only included if HAVE_KMOD is true https://github.com/systemd/systemd/blob/4f44d2c4f76922a4f48dd4473e6abaca40d7e555/src/udev/meson.build#L39 )

Reply:
@lheckemann thanks for the information. I will drop a PR later to but it back on. Do we have any tests covering mobile-nixos use case?

Reply:
Opened #224421, apologies for the inconvenience. A question I have on this: would it make sense to have `systemdMobile` with precisely tailored set of options enabled in order to avoid running into similar problems in the future?